SemanticUI/Modules/Dropdown.elm

```diff
@@ -0,0 +1,456 @@
+        , Dropdown.drawer cfg
+            st
+            div
+            []
+            [ Dropdown.item div [] [ text "New" ]
+            , Dropdown.item div [] [ span [ class "description" ] [ text "ctrl + o" ], text "Open..." ]
+            , Dropdown.item div [] [ span [ class "description" ] [ text "ctrl + s" ], text "Save as..." ]
+            , Dropdown.item div [] [ span [ class "description" ] [ text "ctrl + r" ], text "Rename" ]
+            , Dropdown.item div [] [ text "Make a copy" ]
+            , Dropdown.item div [] [ i [ class "folder icon" ] [], text "Move to folder" ]
+            , Dropdown.item div [] [ i [ class "trash icon" ] [], text "Move to trash" ]
+            , div [ class "divider" ] []
+            , Dropdown.item div [] [ text "Download As.." ]
+            , let
+                cfgSub =
+                    Dropdown.init { identifier = "fileSub", onToggle = ToggleFileSub2 }
+                        |> Dropdown.toggleEvent Dropdown.OnHover
+
+                stSub =
+                    model.fileSub2
+              in
+              Dropdown.root cfgSub
+                stSub
+                (Dropdown.item div)
+                []
+                [ Dropdown.toggle cfgSub stSub div [] [ i [ class "dropdown icon" ] [], text "Publish to Web" ]
+                , Dropdown.drawer cfgSub
+                    stSub
+                    div
+                    []
+                    [ Dropdown.item div [] [ text "Google Docs" ]
+                    , Dropdown.item div [] [ text "Google Drive" ]
+                    , Dropdown.item div [] [ text "Google Drive" ]
+                    , Dropdown.item div [] [ text "Dropbox" ]
+                    , Dropdown.item div [] [ text "Adobe Creative Cloud" ]
+                    , Dropdown.item div [] [ text "Private FTP" ]
+                    , Dropdown.item div [] [ text "Another Service..." ]
+                    ]
+                ]
+            ]
+        ]
+
+-}
+
+import Dropdown
+import SemanticUI.Modules.Common exposing (Render, WrappedNode)
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Json.Decode as Decode
+
+
+{-| Layout function for a drop down.
+
+A function that takes a record with the following functions
+
+  - makeDropdown - the function `root` with `Config` and `State` applied to it.
+  - makeToggle - the function `toggle` with `Config` and `State` applied to it.
+  - makeDrawer - the function `drawer` with `Config` and `State` applied to it.
+  - makeItem - the function `item` with `Config` and `State` applied to it.
+
+The final resultant value is what you decide (some kind of DOM)
+
+-}
+type alias Layout msg html =
+    { makeDropdown : WrappedNode msg
+    , makeToggle : WrappedNode msg
+    , makeDrawer : WrappedNode msg
+    , makeItem : WrappedNode msg
+    }
+    -> html
+
+
+{-| Define when the dropdown should open
+-}
+type ToggleEvent
+    = OnClick
+    | OnHover
+    | OnFocus
```

Comment:
This could be 

```elm
type alias ToggleEvent = Dropdown.ToggleEvent
```

...but I suppose this is necessary in order to re-export the contructors? It's a shame that we have to jump through hoops to do that.